ee/codegen/src/__test__/nodes/__snapshots__/api-node.test.ts.snap

```diff
@@ -94,6 +94,84 @@ class APINode(BaseAPINode):
 "
 `;
 
+exports[`ApiNode > basic api node > should generate API key header fields when API key is provided 1`] = `
+"from vellum.workflows.constants import APIRequestMethod, AuthorizationType
+from vellum.workflows.nodes.displayable import APINode as BaseAPINode
+
+
+class APINode(BaseAPINode):
+    url = "fasdfadsf"
+    method = APIRequestMethod.GET
+    api_key_header_key = "X-API-KEY"
+    authorization_type = AuthorizationType.API_KEY
+    api_key_header_value = "my-api-key"
+"
+`;
+
+exports[`ApiNode > basic api node > should generate all fields when all conditions are provided 1`] = `
+"from vellum.workflows.constants import APIRequestMethod, AuthorizationType
+from vellum.workflows.nodes.displayable import APINode as BaseAPINode
+
+
+class APINode(BaseAPINode):
+    url = "fasdfadsf"
+    method = APIRequestMethod.POST
+    json = {}
+    headers = {
+        "Content-Type": "application/json",
+    }
+    api_key_header_key = "X-API-KEY"
+    authorization_type = AuthorizationType.API_KEY
+    api_key_header_value = "my-api-key"
+"
+`;
+
+exports[`ApiNode > basic api node > should generate body field when body is provided 1`] = `
+"from vellum.workflows.constants import APIRequestMethod
+from vellum.workflows.nodes.displayable import APINode as BaseAPINode
+
+
+class APINode(BaseAPINode):
+    url = "fasdfadsf"
+    method = APIRequestMethod.GET
+    json = {}
```

Comment:
[nit] I'd expect `json = {}` to be skipped, particularly on GET requests. let's:
- update this to a POST example
- add a `"foo": "bar"` to the `json`